### PR TITLE
Update the `GitHub Action` and `AppVeyor` entries on the `developer-tools` page to match the current test content.

### PR DESCRIPTION
**Pull request by LuciferYang**
This pr update the `GitHub Action` and `AppVeyor` entries on the `developer-tools` page to match the current test content.

### CODE REVIEW

File: developer-tools.md [hunk 1]
```diff
@@ -14,18 +14,18 @@ Apache Spark community uses various resources to maintain the community test cov
 <h3 id="github_action">GitHub Action</h3>
 
 [GitHub Action](https://github.com/apache/spark/actions) provides the following on Ubuntu 22.04.
-- Scala 2.12/2.13 SBT build with Java 8
-- Scala 2.12 Maven build with Java 11/17
-- Java/Scala/Python/R unit tests with Java 8/Scala 2.12/SBT
+- Scala 2.13 SBT build with Java 17
```
Comment (LuciferYang) [inline developer-tools.md:20]: cc @wangyum

**Review by wangyum — COMMENTED** [review]
cc @dongjoon-hyun

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Looks fine, only issue is that these docs are meant to be for all current versions, and this info is for 4.0, which isn't released yet. We can do this, but, could also write the docs to have different info for 3.x and 4.x

**Comment by LuciferYang** [comment]
like 
image
?

**Comment by srowen** [comment]
Yep that works. If you think it's simpler you can also just mention these differences inline in the bullet points that changed. IT all works.

**Comment by LuciferYang** [comment]
I have chosen to list them separately so that the parts of Spark 3. x can be deleted at once later. Additionally, it seems that branch-3.x will no longer trigger `AppVeyor`: https://github.com/apache/spark/pull/43676

so there is no longer a distinction between Spark 3. x and good 4. x for AppVeyor

**Review by srowen — APPROVED** [review]

File: developer-tools.md [hunk 2]
```diff
@@ -14,18 +14,31 @@ Apache Spark community uses various resources to maintain the community test cov
...
+- Scala 2.13 Maven build with Java 17/21
+- Java/Scala/Python/R unit tests with Java 17/Scala 2.13/SBT
 - TPC-DS benchmark with scale factor 1
 - JDBC Docker integration tests
 - Kubernetes integration tests
-- Daily Java/Scala/Python/R unit tests with Java 11/17 and Scala 2.12/SBT
+- Daily Java/Scala/Python/R unit tests with Java 21 and Scala 2.13/SBT
+
+<h4 id="spark_3x">Spark 3.x</h4>
```
Comment (dongjoon-hyun) [inline developer-tools.md]: After Apache Spark 3.3 reaches the EOL on December 15th, let's remove this.

**Review by dongjoon-hyun — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

File: site/developer-tools.html [hunk 3]
```diff
@@ -146,21 +146,36 @@ <h2>Test coverage</h2>
 <h3 id="github_action">GitHub Action</h3>
 
 <p><a href="https://github.com/apache/spark/actions">GitHub Action</a> provides the following on Ubuntu 22.04.</p>
+
+<h4 id="spark_4x">Spark 4.x</h4>
```
Comment (HyukjinKwon) [inline site/developer-tools.html]: I would say just sth like "Apache Spark 4 and above" or just "Apache Spark 4" or mention that latest dev branch.

File: site/developer-tools.html [hunk 4]
```diff
@@ -146,21 +146,36 @@ <h2>Test coverage</h2>
...
+  <li>Scala 2.13 Maven build with Java 17/21</li>
+  <li>Java/Scala/Python/R unit tests with Java 17/Scala 2.13/SBT</li>
   <li>TPC-DS benchmark with scale factor 1</li>
   <li>JDBC Docker integration tests</li>
   <li>Kubernetes integration tests</li>
-  <li>Daily Java/Scala/Python/R unit tests with Java 11/17 and Scala 2.12/SBT</li>
+  <li>Daily Java/Scala/Python/R unit tests with Java 21 and Scala 2.13/SBT</li>
+</ul>
+
+<h4 id="spark_3x">Spark 3.x</h4>
```
Comment (HyukjinKwon) [inline site/developer-tools.html]: ditto

**Comment by HyukjinKwon** [comment]
Merged to asf-site.

**Pull request closed without merging**